src/components/Whiteboard/ElementBehaviors/Resizable/ResizeHandle.tsx

```diff
@@ -17,101 +17,127 @@
-  scale: number,
+  params: { scale: number } & (
+    | LineElementProps
+    | PolylineAndShapeElementsProps
+  ),
 ): {
   x: number;
   y: number;
   width: number;
   height: number;
   cursor: string;
 } {
+  const { elementKind, handlePosition, scale } = params;
   const handleWidth = 10 / scale;
   const selectionBorderPadding = 2 / scale;
 
-  switch (position) {
-    case 'top':
-      return {
-        x: handleWidth / 2 - selectionBorderPadding,
-        y: -handleWidth / 2 - selectionBorderPadding,
-        // Make sure we never get negative dimensions, if the element is too small
-        width: Math.max(
-          0,
-          containerWidth + 2 * selectionBorderPadding - handleWidth,
-        ),
-        height: handleWidth,
-        cursor: 'ns-resize',
-      };
-    case 'topRight':
-      return {
-        x: containerWidth - handleWidth / 2 + selectionBorderPadding,
-        y: -handleWidth / 2 - selectionBorderPadding,
-        width: handleWidth,
-        height: handleWidth,
-        cursor: 'ne-resize',
-      };
-    case 'right':
-      return {
-        x: containerWidth + selectionBorderPadding - handleWidth / 2,
-        y: handleWidth / 2 - selectionBorderPadding,
-        width: handleWidth,
-        height: Math.max(
-          0,
-          containerHeight + 2 * selectionBorderPadding - handleWidth,
-        ),
-        cursor: 'ew-resize',
-      };
-    case 'bottomRight':
-      return {
-        x: containerWidth - handleWidth / 2 + selectionBorderPadding,
-        y: containerHeight - handleWidth / 2 + selectionBorderPadding,
-        width: handleWidth,
-        height: handleWidth,
-        cursor: 'se-resize',
-      };
-    case 'bottom':
-      return {
-        x: handleWidth / 2 - selectionBorderPadding,
-        y: containerHeight - handleWidth / 2 + selectionBorderPadding,
-        width: Math.max(
-          0,
-          containerWidth + 2 * selectionBorderPadding - handleWidth,
-        ),
-        height: handleWidth,
-        cursor: 'ns-resize',
-      };
-    case 'bottomLeft':
-      return {
-        x: -handleWidth / 2 - selectionBorderPadding,
-        y: containerHeight - handleWidth / 2 + selectionBorderPadding,
-        width: handleWidth,
-        height: handleWidth,
-        cursor: 'sw-resize',
-      };
-    case 'left':
-      return {
-        x: -handleWidth / 2 - selectionBorderPadding,
-        y: handleWidth / 2 - selectionBorderPadding,
-        width: handleWidth,
-        height: Math.max(
-          0,
-          containerHeight + 2 * selectionBorderPadding - handleWidth,
-        ),
-        cursor: 'ew-resize',
-      };
-    case 'topLeft':
-      return {
-        x: -handleWidth / 2 - selectionBorderPadding,
-        y: -handleWidth / 2 - selectionBorderPadding,
-        width: handleWidth,
-        height: handleWidth,
-        cursor: 'nw-resize',
-      };
+  if (elementKind === 'line') {
+    const { handlePositionX = 0, handlePositionY = 0 } = params;
```

Comment:
No need to assign default 0 to handlePositionX and `handlePositionY`, they are not optional already.